File: duplicateEncoder/duplicateEncoder.py
from collections import Counter

def duplicateEncode(word):
    charCount = Counter(word.lower())
    result = ["(" if charCount[c] == 1 else ")" for c in word.lower() ]
    
    return "".join(result)

if __name__ == "__main__":
    assert(duplicateEncode("din") == "(((");
    assert(duplicateEncode("recede") == "()()()");  
    assert(duplicateEncode("Success") == ")())())");  
    assert(duplicateEncode("(( @") == "))((");
    print("All Test passed!")
    print("-" * 32)

File: duplicateEncoder/duplicateEncoder.js
const assert = require("assert");

function duplicateEncode(word) {
    let charCount = {};
    let result = "";

    word.toLowerCase()
        .split("")
        .forEach((c) => {
            if (c in charCount) {
                charCount[c] = charCount[c] + 1;
            } else {
                charCount[c] = 1;
            }
        });

    word.toLowerCase()
        .split("")
        .forEach((c) => {
            if (charCount[c] == 1) {
                result = result + "(";
            } else {
                result = result + ")";
            }
        });

    return result;
}

describe("Simple Test", () => {
    it("should return (((", () => {
        assert.equal(duplicateEncode("din"), "(((");
    });
    it("should return ()()()", () => {
        assert.equal(duplicateEncode("recede"), "()()()");
    });
    it("should return )())())", () => {
        assert.equal(duplicateEncode("Success"), ")())())");
    });
    it("should return ))((", () => {
        assert.equal(duplicateEncode("(( @"), "))((");
    });
});
